addons/mail/static/src/core/common/message.js

```diff
@@ -416,6 +416,8 @@ export class Message extends Component {
                     res_model: model,
                     views: [[false, "form"]],
                     res_id: id,
+                }).then(() => {
+                    this.threadService.open(this.props.thread);
```

Comment:
We might want to directly ask to open a chat window, rather than waiting for the action to execute and only then using thread open, hoping it would open the chat window.

Reply:
I did that to avoid showing the chat window when we already are on a conversation. Isn't that annoying to have the chat window and the conversation opened at the same time? Mmmh... Wait... When we call the thread service, it's not certain that the chat window will open? Why?

Reply:
If discuss app is (still) open it opens in discuss app instead.

The chat window is automatically hidden in discuss app.

Reply:
Ok, so it's better to wait for the action service to complete before opening the chat window. When the promise returned by the action service is resolved, I assume that the view has been loaded properly. Therefore, awaiting the promise is necessary and ensure that the conversation won't be opened in the discuss app.